File: 05_이원영/week2/1085.js
//9-7 1085
const input = [6, 2, 10, 3];
const x = input[0];
const y = input[1];
const w = input[2];
const h = input[3];
const counters = [x, y, w - x, h - y];
 
console.log(Math.min.apply(null, counters));

File: 08_전세현/week2/1085.py
import sys
input=sys.stdin.readline

x,y,w,h=map(int,input().split())

if w>1 and h <=1000:
    if 1<=x<=w-1 and 1<=y<=h-1:
        print(min([x,y,w-x,h-y]))